src/Control/Monad/Free.hs

```diff
@@ -53,14 +51,24 @@ import Control.Monad.State.Class
 import Control.Monad.Error.Class
 import Control.Monad.Cont.Class
 import Data.Functor.Bind
+import Data.Functor.Classes
 import Data.Foldable
 import Data.Profunctor
 import Data.Traversable
 import Data.Semigroup.Foldable
 import Data.Semigroup.Traversable
 import Data.Data
 import Prelude hiding (foldr)
-import Prelude.Extras
+
+import qualified Prelude.Extras as PE
```

Comment:
Is your goal to remove all uses of `prelude-extras` in this PR? Or is that planned for later?